lightning/src/onion_message/packet.rs

```diff
@@ -314,7 +314,7 @@ impl Readable for ControlTlvs {
 			(None, None) => None,
 		};
 
-		let valid_fwd_fmt = next_hop.is_some() && path_id.is_none();
```

Comment:
Should this check stay?

Reply:
I think we should keep this check. It's better to have a (potentially redundant?) check than to miss something important, right? :)
I've reverted the check to its original version in [8e4e1aa](https://github.com/lightningdevkit/rust-lightning/commit/8e4e1aabadb5fda9f4d0ec2c845263c94c273643).

Let me know what you think!

Reply:
Understood! Kept the check intact in the current version of PR.